### PR TITLE
MLOPS-346 Separate Ambassador module from Seldon module

### PR DESCRIPTION
**Pull request by bernardolk**

### CODE REVIEW

**Comment by bernardolk** [comment]
Needs to be tested and a review

File: main.tf [hunk 1]
```diff
@@ -145,6 +145,18 @@ module "seldon" {
...
+resource "kubernetes_namespace" "ambassador_namespace" {
+  count = var.ambassador_enabled ? 1 : 0
+  metadata {
+    name = var.ambassador_namespace
+  }
+}
+
+module "ambassador" {
+  count = var.ambassador_enabled ? 1 : 0
+  source    = "./modules/ambassador"
```
Comment (tvinagre) [inline main.tf:159]: Would be good to add `namespace` here as well

File: modules/ambassador/main.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+resource "helm_release" "ambassador" {
+  count = var.ambassador_enabled ? 1 : 0
+  repository = "https://www.getambassador.io"
+  chart = "ambassador"
+  name = "ambassador"
+  namespace = var.ambassador_namespace
```
Comment (tvinagre) [inline modules/ambassador/main.tf:6]: could you rename this var to `namespace` please? So it gets consistent with the rest of the repo

File: modules/ambassador/main.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (tvinagre) [inline modules/ambassador/main.tf:7]: Also, I'm missing a HELM version here, would be good to add to prevent future issues

File: main.tf [hunk 4]
```diff
@@ -145,6 +145,18 @@ module "seldon" {
   count = var.install_seldon ? 1 : 0
   source    = "./modules/seldon"
   namespace = kubernetes_namespace.seldon_namespace[0].metadata[0].name
+}
+
+resource "kubernetes_namespace" "ambassador_namespace" {
```
Comment (tvinagre) [inline main.tf:150]: You fixed the typo, nice :)

File: modules/seldon/variables.tf [hunk 5]
```diff
@@ -1,40 +1,8 @@
-variable "namespace" {
+variable "seldon_namespace" {
```
Comment (tvinagre) [inline modules/seldon/variables.tf:1]: Please let's just keep `namespace`. Also, if you rename it here, you need to change on `main.tf`, where the `module` gets called

**Review by tvinagre — CHANGES_REQUESTED** [review]
Awesome! Thanks @bernardolk just some small suggestions. Let's just test is on the cluster before merging :)

File: main.tf (hunk 4 above)
Comment (bernardolk) [reply main.tf:150]: 🦾

**Pull request merged**